skyportal/plot.py

```diff
@@ -162,6 +162,72 @@ def get_color(bandpass_name):
         return bandcolor
 
 
+def annotate_spec(plot, spectra, lower, upper):
+    """Annotate photometry plot with spectral markers that contain hover info
+    Parameters
+    ----------
+    plot: bokeh figure object
+        Figure to be annotated
+    spectra: DBSession object
+        Results of query for spectra of object
+    lower: float
+    upper: float
+        Plot limits allowing calculation of annotation symbol y value
+    """
+    s_mjd = []
+    s_y = []
+    s_date = []
+    s_tel = []
+    s_inst = []
+    s_text = []
+    text_y = upper - (upper - lower) * 0.05
+    for s in spectra:
+        date = '%d-%d-%dT%02d:%02d:%02d' % (
+            s.observed_at.year,
+            s.observed_at.month,
+            s.observed_at.day,
+            s.observed_at.hour,
+            s.observed_at.minute,
+            s.observed_at.second,
+        )
+        s_mjd.append(Time(date).mjd)
+        s_y.append(text_y)
+        s_date.append(date)
+        s_tel.append(s.instrument.telescope.name)
+        s_inst.append(s.instrument.name)
+        s_text.append("S")
+    if len(s_mjd) > 0:
+        spec_r = plot.text(
+            x='s_mjd',
+            y='s_y',
+            text='s_text',
+            text_alpha=0.3,
+            text_align='center',
+            source=ColumnDataSource(
+                data=dict(
+                    s_mjd=s_mjd,
+                    s_y=s_y,
+                    s_date=s_date,
+                    s_tel=s_tel,
+                    s_inst=s_inst,
+                    s_text=s_text,
```

Comment:
```suggestion
                    s_mjd=[Time(s.observed_at, format='datetime').mjd for s in spectra],
                    s_y=[text_y] * len(spectra),
                    s_date=[s.observed_at.isoformat() for s in spectra],
                    s_tel=[s.instrument.telescope.name for s in spectra],
                    s_inst=[s.instrument.name for s in spectra],
                    s_text=['S'] * len(spectra),
```

Reply:
Done, minus the extraneous commas and organized by calculated and spectral data.